### PR TITLE
Change CredentialLocation and Vault fields in DataStore struct to opt…

### PR DESCRIPTION
**Pull request by revit13**

### CODE REVIEW

**Comment by roee88** [comment]
You need to run make generate and make manifests

File: manager/apis/app/v1alpha1/datacatalog_types.go [hunk 1]
```diff
@@ -12,10 +12,10 @@ import (
 type DataStore struct {
 	// CredentialLocation is used to obtain
 	// the credentials from the credential management system - ex: vault
-	// +required
+	// +optional
 	CredentialLocation string `json:"credentialLocation"`
 	// Holds details for retrieving credentials by the modules from Vault store.
-	// +required
+	// +optional
```
Comment (froesef) [inline manager/apis/app/v1alpha1/datacatalog_types.go:18]: If a struct is optional it should be a pointer. 
=> ```Vault *Vault `json:"vault,omitempty"````

File: manager/apis/app/v1alpha1/datacatalog_types.go [hunk 2]
```diff
@@ -12,10 +12,10 @@ import (
 type DataStore struct {
 	// CredentialLocation is used to obtain
 	// the credentials from the credential management system - ex: vault
-	// +required
+	// +optional
```
Comment (froesef) [inline manager/apis/app/v1alpha1/datacatalog_types.go:15]: Please add omitempty flag.

Comment (revit13) [reply manager/apis/app/v1alpha1/datacatalog_types.go:15]: Sure Done.

File: manager/apis/app/v1alpha1/datacatalog_types.go (hunk 1 above)
Comment (revit13) [reply manager/apis/app/v1alpha1/datacatalog_types.go:18]: Thanks, I fixed that.

**Review by froesef — APPROVED** [review]

**Pull request merged**